modules/ROOT/pages/custom-ingress-configuration.adoc

```diff
@@ -2,14 +2,29 @@
 
 [NOTE]
 ====
-The tasks outline in this topic must be performed by an IT administrator
+The tasks outlined in this topic must be performed by an IT administrator.
 ====
 
-Runtime Fabric on Self-Managed Kubernetes enables you to edit a template file that specifies a custom Ingress configuration. Using this file you can define values for annotations, rewrite rules, and the ingressClassName.
+Runtime Fabric on Self-Managed Kubernetes enables you to edit a template file that specifies a custom Ingress configuration. Using this file you can define values for annotations, rewrite rules, and the `ingressClassName`.
 
-The template file is a Kubernetes ConfigMap file. After customizing the template, you can use the `rtfctl apply ingress-configmap` command-line utility to apply the custom Ingress configuration to your Kubernetes cluster.
+The template file is a Kubernetes ConfigMap file. After customizing the template, use the `rtfctl apply ingress-configmap` command to apply the custom Ingress configuration to your Kubernetes cluster.
 
-If you do not apply a custom ConfigMap file based on a template file, Runtime Fabric will default to a standard ingress resource definition. Depending on your ingress controller, this resource definition may not work. In this case, you must apply a custom ConfigMap using the template.
+If you do not apply a custom ConfigMap file based on a template file, Runtime Fabric defaults to a standard ingress resource definition. Depending on your ingress controller, this resource definition may not work. In this case, you must apply a custom ConfigMap using the template.
+
+== How Ingress Resources are Generated
+
+When deploying a Mule application to Runtime Fabric on Self-Managed Kubernetes, the Runtime Fabric agent generates a service and ingress resource for that application. The Runtime Fabric agent creates these Kubernetes resources in a namespace that corresponds to the Anypoint Environment ID used when deploying the Mule application.
+
+By default, each application deployment has its own service and ingress resources. 
+
+In Runtime Fabric v1.8.50 and later, annotations and labels can be added to the generated Service resources. These changes persist for the lifecycle of those resources.
```

Comment:
@sarjeet2013 Could we update this with the right agent version? (I just put down v1.8.50 as a placeholder).

Reply:
v1.8.50 is correct.